test/fixtures/dialects/ansi/merge_into.yml

```diff
@@ -231,7 +231,7 @@ file:
       - merge_when_matched_clause:
         - keyword: WHEN
         - keyword: MATCHED
-        - binary_operator: AND
+        - keyword: AND
```

Comment:
Might as well fix then. Easy enough to update `MergeMatchedClauseSegment` to replace `"AND"` with `Ref("AndGrammar")`. But need to just check other dialects too and then regen YAMLs.

But does that mean `WHEN MATCH && ...` is valid? Because that change will allow that. And it no, then is it really a binary operator?

Reply:
Actually not 100% sure on this.

https://docs.snowflake.com/en/sql-reference/sql/merge.html

> WHEN MATCHED [ AND <case_predicate> ] THEN

So the `AND` is specific syntax for the case predicate and not a Boolean operator (that could be replaced with `OR` for example). So I think it should be a keyword here too.

Reply:
Yes, as I suppose you remembered, a few others like BigQuery should be made consistent in inheriting classes like `MergeNotMatchedBySourceClauseSegment`. I'll look at it soon.

Reply:
Yeah but it’s back to being a type  `boolean_operator` (which admittedly it was before this PR) when I’m not convinced it should be.

And if MySQL ever did introduce a `MERGE` statement, SQLFluff would `AND` or `&&` if we use `AndOperatorGrammar` when it likely really should just be `AND`.

Reply:
Alright, I can buy that.
Equivalently, MySQL certainly does not accept `create || replace view ...```